lighthouse-cli/printer.ts

```diff
@@ -17,45 +17,69 @@
 
 'use strict';
 
-const fs = require('fs');
-const ReportGenerator = require('../lighthouse-core/report/report-generator');
-const Formatter = require('../lighthouse-core/formatters/formatter');
-const log = require('../lighthouse-core/lib/log');
-
 /**
  * An enumeration of acceptable output modes:
- * <ul>
- *   <li>'pretty': Pretty print the results</li>
- *   <li>'json': JSON formatted results</li>
- *   <li>'html': An HTML report</li>
- * </ul>
- * @enum {string}
+ *   'pretty': Pretty print the results
+ *   'json': JSON formatted results
+ *   'html': An HTML report
  */
-const OUTPUT_MODE = {
-  pretty: 'pretty',
-  json: 'json',
-  html: 'html'
+enum OutputMode { pretty, json, html };
+type Mode = 'pretty' | 'json' | 'html';
+const Modes = ['pretty', 'json', 'html'];
+
+interface SubScore {
+  displayValue: string;
+  debugString: string;
+  score: number;
+  description: string;
+  extendedInfo?: {
+    value: string;
+    formatter: string;
+  };
+}
+
+interface Score {
+  overall: number;
+  name: string;
+  scored: boolean;
+  subItems: Array<SubScore | string>;
+}
+
+interface Aggregation {
+  name: string;
+  score:  Array<Score>;
+}
+
+interface Results {
+  url: string;
+  aggregations: Array<Aggregation>;
+  audits: Object;
+  lighthouseVersion: string;
 };
 
+const fs = require('fs');
+const ReportGenerator = require('../lighthouse-core/report/report-generator');
+const Formatter = require('../lighthouse-core/formatters/formatter');
+const log = require('../lighthouse-core/lib/log');
+
 /**
  * Verify output mode.
- * @param {string} mode
- * @return {OUTPUT_MODE}
  */
-function checkOutputMode(mode) {
-  if (!OUTPUT_MODE.hasOwnProperty(mode)) {
-    log.warn('Printer', `Unknown output mode ${mode}; using pretty`);
-    return OUTPUT_MODE.pretty;
+function checkOutputMode(mode: Mode): OutputMode {
+  let modeIndex = Modes.indexOf(mode);
+
+  if (modeIndex === -1) {
+      log.warn('Printer', `Unknown output mode ${mode}. using pretty`);
+      return OutputMode.pretty;
   }
 
-  return OUTPUT_MODE[mode];
+  return modeIndex;
```

Comment:
we have type checks on 68 for the input and output of this fn.. we also have an indexOf check to ensure it matches our enum values. so it is not really tenuous :)